### ISSUE TITLE
Pressing space key while focused on the "permanently delete" checkbox in the "delete item(s)" modal dialog breaks keyboard navigation

### ISSUE THREAD

**Issue by stamminator** [files-community/Files#1323]
**Describe the bug**
When the modal dialog for deleting a folder/file pops up, pressing `Space` while focused on the "Permanently delete" checkbox causes the focus to leave the dialog and go back to the main window, even though the dialog is still present. There appears to be no way to return focus to the dialog window with keyboard navigation alone, forcing the user to use the cursor.

**To Reproduce**
1. Right click, then select Delete on a file/folder, or press the `Delete` key.
2. Press `Tab` to set focus on the "permanently delete" checkbox field.
3. Press `Space`.
4. Observe that the focus has now been moved to the main window instead of the modal.

**Expected behavior**
1. Pressing the space key should toggle the checked status of the checkbox.
2. It should be impossible for a modal to lose focus and still be visible. Losing focus should dismiss the modal.

**Screenshots**

![sZ0XYLcr5J](https://user-images.githubusercontent.com/10633170/86276969-e5f01000-bba3-11ea-9f67-05f1cfa54cc3.gif)

**Desktop:**
 - OS Version: Windows 10 2004 19041.329
 - App version: v0.11.0

**Comment by gave92** [comment]
Guess this might be related to the method `Page_CharacterReceived` in `GenericFileBrowser` where we have the line `AllView.Focus(FocusState.Keyboard);` that steals the focus from the displayed dialog. We should do this only if no dialog is displayed.

**Comment by Thomas1664** [comment]
@yair100 Since this issue is fixed, you can go ahead and close this.

**Comment by yaichenbaum** [comment]
@stamminator Can you confirm if this is fixed on your end?

**Comment by stamminator** [comment]
@yair100 Well it appeared to be fixed in the initial release that contained the fix several weeks ago, but the .16 and .17 builds are unresponsive for me, so I can't confirm.

EDIT: It looks like all functionality except for the fancy cursor hover effects is completely frozen for about the first 40 seconds after startup, as if it's waiting for some synchronous task to complete. Once I waited for it to become responsive, I was able to confirm the modal issue no longer takes place.
